compiler/damlc/daml-opts/daml-opts/DA/Daml/Options.hs

```diff
@@ -68,6 +71,21 @@ toCompileOpts options@Options{..} reportProgress =
                 then Just path
                 else Nothing
       | otherwise = pure Nothing
+    -- NOTE(MH): The type class contexts used with generic templates sometimes
+    -- trigger the `Opt_WarnSimplifiableClassConstraints` warning, which
+    -- suggests to use the `MonoLocalBinds` language extension. However, this
+    -- extensions does no play well with our record-dot-syntax. Thus, we
+    -- suppress these warnings.
+    isGenTemplMonoLocal msg
```

Comment:
It would be nice to add an example of the full error message here to make it clear what we are trying to match against.